02-getting_started.Rmd

```diff
@@ -1,67 +1,135 @@
+The `r-devel` at [R/trunk](trunk "svn devel source code folder") is the next minor or eventual major release development version of R.
+Mostly, bug fixes are introduced in `r-patched` and `r-devel`, while `r-devel` is for introducing new features.
 
-2. Click on the download links to download an executable installer.
+## Installing R in Linux
 
-3. Select the language while installing, read the GNU general public license information, and select the destination location to start the installation. You will be prompted to select components at this stage: `User installation`, `32-bit User installation`, `64-bit User installation`, or `Custom installation`. The default option may be chosen for the questions from this step onwards to complete the installation. 
+Here are the basic steps needed meant as a checklist.For complete instructions please see the section in [R-admin](https://cran.r-project.org/doc/manuals/r-devel/R-admin.html#Installing-R-under-Unix_002dalikes)
 
-## Building R and R packages
+1.  Retrieve R source code via into `TOP_SRCDIR`, note that we retrieve the r-devel source code:
 
-### What tools do you need to build R from source on Windows?
+        svn checkout https://svn.r-project.org/R/trunk/ TOP_SRCDIR
 
-1. [RTools](https://github.com/r-windows/docs/blob/master/faq.md#what-is-rtools) is the toolchain bundle that you can use to build R base and R packages containing compiled code, on Windows.
+    If you need to install svn you can use your distribution's package manager to install it.
+    In Ubuntu you can use :
 
-2. You also need a distribution of LaTeX installed for building R and checking packages. The MiKTeX distribution of LaTeX that is used on CRAN can be downloaded from https://miktex.org.
+        sudo apt-get install subversion
 
-### How to setup `RTools`?
+2.  Download the latest recommended packages[^02-getting_started-2]:
 
-1. The latest version of `RTools` can be downloaded from https://cran.r-project.org/bin/windows/Rtools/ and run in the Windows-style installer. You will need to know if you have a 32-bit or 64-bit Windows machine (right-click `This PC` in Windows Explorer and check the properties if you are unsure).
+        TOP_SRCDIR/tools/rsync-recommended
 
-2. Don't forget to add `RTools` to the path as documented on the download page.
+3.  Configure indicating where R should be installed (`BUILDDIR`):
 
-### How to build R?
+        TOP_SRCDIR/configure –prefix=BUILDDIR
 
-To build R for Windows using `RTools` follow the instructions in this [README](https://github.com/r-windows/r-base#readme) file. There are two options available to build R. One is the quick development build and the other option is the full installer build.
+    or
 
-For development and testing, you need only the quick development build. The quick build avoids building the manuals, which are generally not needed for development and testing. 
+        cd BUILDDIR
+        TOP_SRCDIR/configure
```

Comment:
I will correct and look for -- and not a em-dash

Reply:
I like the idea of using TOP_SRCDIR=$HOME/Downloads/R since contributors will only ever be "downloading" from the repository.

I agree using BUILDDIR=$HOME/bin/R-devel should be safer than making an `R-devel` directory in the users home directory.

This discussion supports the idea of using variables though, as people might want to adjust these locations to their preference and if we define them up front that should be clear and simple.

It is not necessary to install R to use it with RStudio. I doubt this is an issue for Emacs either, not sure about other IDEs.

I think since we are writing these instructions for individual contributors (not sys admins) and we don't want to interfere with any existing installation, then we don't need to run `make install`. If we did then /opt/R/R-devel would be a good place to install I think, but I can't see any advantage in doing this (we can check about this).

I think we should add to the instructions to delete all files in BUILDDIR before rebuilding as I think not doing this might cause problems like you had.

I think this is the right place to mention configuration options like the one for RStudio as it is annoying if people follow the instructions then find they have to do it all again (or most of it) to use with RStudio.

Reply:
Also as we are not necessarily installing R, perhaps this section should be called "Building R on Linux".

Reply:
Ok, I will change and add the variables and add the RStudio required option for the users